### ISSUE TITLE
[QUESTION] I need better doc or help with Custom Element

### ISSUE THREAD

**Issue by yespbs** [laraform/laraform#58]
### Prerequisites
Custom element development

* [x] You've read the [documentation](http://laraform.io/docs) and couldn't find an answer.
* [x] Your question isn't already asked before.
* [x] You filled in the entire issue template.

### Versions
1.1.7 premium

* PHP version: 
* Laravel version: 
* Laraform Vue package version and type: 
* Laraform Laravel package version: 

### Description

I am trying to create a custom "signature" element using this vue plugin https://www.npmjs.com/package/vue-signature-pad. While this is fairly basic and works in a preview mode, fails to load in a PWA environment. Preview mode is just simple Laravel Spark Page with part vue js ( not SPA ). I followed https://laraform.io/docs/1.x/how-tos/creating-element as well core code to draft the idea. No clue why it fails to render on PWA

Could you prepare a more detailed custom element development procedure? I need to render and capture the output ( not a model like TEXT element, a callback or hook to invoke custom code for example.

### Additional Information
Screen shots attached, one from Preview and one from PWA
![editor_signature_element](https://user-images.githubusercontent.com/478293/73719838-94536e80-4746-11ea-94cb-2b429dcf1203.png)
![pwa_signature_element](https://user-images.githubusercontent.com/478293/73719839-94ec0500-4746-11ea-82d1-a0834d2cb755.png)

Any additional information, configuration or data that might be necessary to reproduce the issue.

**Comment by yespbs** [comment]
Json Schema for larafom 

`Array
(
    [key] => Array
        (
            [campaign_uuid] => 780d3b0f-8756-488e-a571-fa843ffb64f1
            [form_uuid] => a4ff99c7-3d26-49c2-805b-fcd20464fee0
        )

    [schema] => Array
        (
            [Your Name] => Array
                (
                    [type] => text
                    [label] => Your Name
                    [inputType] => text
                    [name] => Your Name
                    [rules] => Array
                        (
                        )

                )

            [Signature] => Array
                (
                    [type] => signature
                    [label] => Signature
                    [name] => Signature
                )

        )

)`

**Comment by laraform** [comment]
Hi @yespbs! What do you need help with exactly? The error code says that there's no template defined for the component, which could be for couple of reasons, but without seeing the whole code it's hard to tell. Probably that's not an issue with Laraform if you've followed the steps in the Custom element page, but something in the middle.

To make such thing work I'd recommend checking out how Trix for example is implemented. There's a TrixElement which contains a Trix component and they communicate. In your case you should have a SignatureElement which should contain this VueSignaturePad.

Every element has some common features and props, see here:
https://laraform.io/docs/1.x/reference/frontend-element

In theory VueSignaturePad emits an event when the signature is finished, then the SignatureElement component listens to it and updates its data which can be submitted as part of the form's data.

**Comment by yespbs** [comment]
Hi @laraform The template exists and its perfectly displayed in a Laravel Spark layout using <laraform> component i.e. <laraform :form="{!! $mySchema !!}"></laraform> mySchema being a json schema.

But problem arises when the same component used in a PWA ( not SPA) and throws "template not found" its weird. The component I created is pretty basic to test the feasibility. 
I have attached the screenshot of the component. I am anyway looking at Trix if I get some help. If not I will revert back. 
![Screenshot (25)](https://user-images.githubusercontent.com/478293/74039600-229a4f80-49e8-11ea-9ec0-dbe01df3982a.png)

**Comment by laraform** [comment]
I'd recommend is trying to create a very simple custom element (like an `input` field) first based on the steps described here:
https://laraform.io/docs/1.x/how-tos/creating-element

After that implementing that Signature should be more straightforward. The reason it throws this error is unclear to me just like what you mean by PWA exactly.

**Comment by yespbs** [comment]
Well I somehow managed it, used HiddenElement to extend the new component and push the generated base64data to hidden value when signature drawn.

The actual error that is published here was due to incorrect import, as I used global register, only documented import style works.
Works....
`import SignatureElement from './lib/elements/SignatureElement.vue'
Laraform.element('signature', SignatureElement)`

Thanks for your guidance. just updated the issue in case some one stuck like me :)
